mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -741,6 +745,34 @@ static void reconfigureLAGeneric(LinalgAlignRewriter &b,
   laGeneric.setIteratorTypesAttr(ArrayAttr::get(ctx, iteratorTypes));
 }
 
+static LogicalResult canFuseAcrossAtomic(LinalgAlignRewriter &b,
+                                         linalg::GenericOp laGeneric) {
+  bool isLegal = true;
+  for (auto &region : laGeneric->getRegions()) {
+    for (auto &block : region) {
+      for (auto &op : block) {
+        bool isTruncOp = llvm::isa<arith::TruncFOp, arith::TruncIOp>(op);
+        if (isTruncOp) {
```

Comment:
1. `trunci` needs a separate legality test - I don't think we're set up for atomics of i16 or less
2. Style wile, instead of those `isTruncOp`s, I'd've written
    ```c++
    if (auto truncfOp = dyn_cast<arith::TruncFOp>(op)) {
      ...
    }
    else if (auto truncIOp = dyn_cast<arith::TruncIOp>(op)) { ... }
    else if (isa<linalg::YieldOp>) { ... }
    else { isLegal = false; break; }
    ````
     (up to my hasty indentation)

Reply:
Good point. I like the style. One thing that I change is that I removed `break` from all branches and added the following to the end of the inner loop

```cpp
        if (!isLegal)
          break;
```